Refs #23919 -- Removed Python 2 proxied methods on File.

jschneier opened the pull request:
I noticed this when looking at 98ee57e343206ef553de78b22be5e9a6dacb5060 again. Both `xreadlines` and `softspace` used to exist on file objects but were removed in Python 3. From taking a critical look at what counts as a file I'd say `.newlines` is close but I have left it for now, these 2 should be non-controversial.

timgraham commented:
I don't think it's best to remove `HttpRequest.xreadlines()` without a deprecation considering that method works fine on Python 3.

jschneier commented:
You're probably right, I updated my patch.

charettes commented on django/http/request.py:
```diff
@@ -328,14 +330,20 @@ def readline(self, *args, **kwargs):
+            'request directly.', RemovedInDjango30Warning
+        )
+        for line in self:
+            yield line
```
use `yield from self`.

jschneier replied:
done

timgraham commented on docs/spelling_wordlist:
```diff
@@ -855,6 +856,7 @@ wsgi
 www
 xe
 xgettext
+xreadlines
```
Shouldn't be needed in the word list as long as all instances of this are surrounded by backticks.

timgraham commented on docs/spelling_wordlist:
```diff
@@ -14,6 +14,7 @@ alphanumerics
 amet
 amongst
 analytics
+api
```
shouldn't be needed in the word list-- use API in the docs instead

timgraham commented on docs/ref/request-response.txt:
```diff
@@ -378,7 +385,6 @@ Methods
         for element in ET.iterparse(request):
             process(element)
 
-
```
please don't make unrelated whitespace changes

jschneier commented:
Addressed all concerns

> On Apr 26, 2017, at 12:23, Tim Graham <notifications@github.com> wrote:
> 
> @timgraham commented on this pull request.
> 
> Please move the deprecation to a separate PR and create a ticket for it. See also the deprecation checklist.
> 
> In docs/spelling_wordlist:
> 
> > @@ -855,6 +856,7 @@ wsgi
>  www
>  xe
>  xgettext
> +xreadlines
> Shouldn't be needed in the word list as long as all instances of this are surrounded by backticks.
> 
> In docs/spelling_wordlist:
> 
> > @@ -14,6 +14,7 @@ alphanumerics
>  amet
>  amongst
>  analytics
> +api
> shouldn't be needed in the word list-- use API in the docs instead
> 
> In docs/ref/request-response.txt:
> 
> > @@ -378,7 +385,6 @@ Methods
>          for element in ET.iterparse(request):
>              process(element)
>  
> -
> please don't make unrelated whitespace changes
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

jschneier commented:
Ah need to create a separate ticket for the deprecation.

> On Apr 26, 2017, at 12:23, Tim Graham <notifications@github.com> wrote:
> 
> @timgraham commented on this pull request.
> 
> Please move the deprecation to a separate PR and create a ticket for it. See also the deprecation checklist.
> 
> In docs/spelling_wordlist:
> 
> > @@ -855,6 +856,7 @@ wsgi
>  www
>  xe
>  xgettext
> +xreadlines
> Shouldn't be needed in the word list as long as all instances of this are surrounded by backticks.
> 
> In docs/spelling_wordlist:
> 
> > @@ -14,6 +14,7 @@ alphanumerics
>  amet
>  amongst
>  analytics
> +api
> shouldn't be needed in the word list-- use API in the docs instead
> 
> In docs/ref/request-response.txt:
> 
> > @@ -378,7 +385,6 @@ Methods
>          for element in ET.iterparse(request):
>              process(element)
>  
> -
> please don't make unrelated whitespace changes
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

jschneier commented:
Okay I have updated this PR to not handle the deprecation and will open a new ticket + PR
